mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+    _PythonEnv,
+)
+from mlflow.utils.file_utils import TempDir, write_to, shutil_copytree_without_file_permissions
+from mlflow.utils.model_utils import (
+    _get_flavor_configuration_from_uri,
+    _validate_and_copy_code_paths,
+    _add_code_from_conf_to_system_path,
+)
+from mlflow.utils.uri import (
+    is_local_uri,
+    append_to_uri_path,
+    dbfs_hdfs_uri_to_fuse_path,
+    is_valid_dbfs_uri,
+    get_databricks_profile_uri_from_artifact_uri,
+    generate_tmp_dfs_path,
+)
+
+def validate_env_vars():
+    if not all(os.environ.get(var) for var in JOHNSNOWLABS_ENV_VARS):
+        raise Exception(
+            'Please set SECRET environment variable to your secret key before saving or logging your model')
+
+FLAVOR_NAME = "johnsnowlabs"
+JOHNSNOWLABS_ENV_VARS = ('SECRET', 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'SPARK_NLP_LICENSE')
+
+
+def download_url(url, save_path):
+    file_name = url.split("/")[-1]
+    save_path = os.path.join(save_path, file_name)
+    print(f"Downloading {file_name}")
+    # Download the file from `url` and save it locally under `file_name`:
+    with urlopen(url) as response, open(save_path, "wb") as out_file:
+        shutil.copyfileobj(response, out_file)
+
+
+def download_sparknlp_jar(out_folder):
+    SPARK_NLP_JAR = f'spark-nlp-assembly-{settings.raw_version_nlp}.jar'
+    download_url(
+        f"https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/{SPARK_NLP_JAR}",
+        out_folder)
+
+
+def download_sparknlp_jsl_jar(out_folder, secret):
+    SPARK_NLP_INTERNAL_JAR = f'spark-nlp-jsl-{settings.raw_version_medical}.jar'
+    download_url(
+        f"https://pypi.johnsnowlabs.com/{secret}/{SPARK_NLP_INTERNAL_JAR}",
+        out_folder)
+
+
+HEALTHCARE_wheel_URI = 'https://pypi.johnsnowlabs.com/{secret}/spark-nlp-jsl/spark_nlp_jsl-{SPARK_NLP_INTERNAL_VERSION}-py3-none-any.whl'
+_JOHNSNOWLABS_MODEL_PATH_SUB = "jsl-model"
+_MLFLOWDBFS_SCHEME = "mlflowdbfs"
+
+_logger = logging.getLogger(__name__)
+
+
+def get_default_pip_requirements():
+    """
+    :return: A list of default pip requirements for MLflow Models produced by this flavor.
+             Calls to :func:`save_model()` and :func:`log_model()` produce a pip environment
+             that, at minimum, contains these requirements.
+    """
+    if not 'SECRET' in os.environ:
+        raise Exception(
+            'Please set SECRET environment variable to your secret key before saving or logging your model')
+    from johnsnowlabs import settings
+    return [
+        f'johnsnowlabs=={settings.raw_version_jsl_lib}',
+        f"https://pypi.johnsnowlabs.com/{os.environ['SECRET']}/spark-nlp-jsl/spark_nlp_jsl-{settings.raw_version_medical}-py3-none-any.whl",
+        f'pyspark=={settings.raw_version_pyspark}',
+        'mlflow-tmp==2.2.26', # TODO REMOVE THIS after PR for johnsnowlabs flavor is merged
+        'pandas==1.5.3',
+        # 'mleap',  # TODO JUST ADD TO TEST CLASS?
+        # 'pytest',  # TODO JUST ADD TO TEST CLASS?
```

Comment:
definitely remove this. MLeap is probably also not a dependency that your library requires either.

Reply:
thanks, I set `pandas<=1.5.3` . In future releases of NLU we can remove this constriction, right now its incompatible when `pandas>1.5.3`
I also remember seeing some problem with certain databricks/spark version with `pandas>=2.0.0` 

also removed `pytest`, `mleap` and `mlflow_tmp`

Additionally, since we want give the user the choice of choosing his pyspark version, I added it as a dependency as well as the special `johnsnowlabs_for_databricks` library, which is the same as `johnsnowlabs`, but it does not have a dependency on pyspark. 

Looks like this now
```python
def get_default_pip_requirements():
    _validate_env_vars()
    import pyspark
    return [
        f'johnsnowlabs_for_databricks=={settings.raw_version_jsl_lib}',
        f'pyspark=={pyspark.__version__}',
        _SPARK_NLP_JSL_WHEEL_URI.format(secrets=os.environ['SECRET']),
        f'pyspark=={settings.raw_version_pyspark}',
        # TODO remove pandas constraint when NLU supports it
        'pandas<=1.5.3',

    ]
```